### DISCUSSION TITLE
Allow the location for gatsby-types.d.ts to be configured

### DISCUSSION THREAD

**Discussion by nrandell** [gatsbyjs/gatsby#35779] [Ideas / Feature Requests] [answered]
It would be great to be able to provide a configuration option to write the gatsby-types.d.ts in another location. There are a few reasons for doing this

- in case something else detects the file being created and creates an infinite loop in the build (eg tailwind)
- when you don't want that file added to git and keeping it out of the src folder is useful

Using the new file is a simple matter of adding it to tsconfig.json and everything would continue to work correctly.

**Comment by zhaohuanyuu** [comment]
i have the same question(tailwind loop), hope solved soon

**Comment by LekoArts** [comment, marked as answer]
PR for it is up now: https://github.com/gatsbyjs/gatsby/pull/35966

You can try this now with `gatsby@next` or wait for 4.18 release.
